base/error.jl

```diff
@@ -70,7 +70,7 @@ function _reformat_bt(bt, bt2)
     i, j = 1, 1
     while i <= length(bt)
         ip = bt[i]::Ptr{Cvoid}
-        if ip == Ptr{Cvoid}(-1%UInt)
+        if UInt(ip) == (-1 % UInt)
```

Comment:
So the issue is that we make this `-1` Ptr into a literal and then don't serialize it properly. That seems ... bad?

Reply:
Yes, we save all pointers as NULL so that we can detect that an object is uninitialized at run time, and re-create the needed native objects. Interestingly dump.c has this:
```
                if (t->name == jl_pointer_typename && jl_unbox_voidpointer(v) != (void*)-1) {
                    // normalize most pointers to NULL, to help catch memory errors
                    // but permit MAP_FAILED / INVALID_HANDLE to be stored unchanged
```
but staticdata.c doesn't have the special case for -1. @vtjnash & I discussed limiting this somehow, e.g. only nulling mutable object fields.

Reply:
This one is particularly bad because the user didn't write `@eval :($(Ptr{Cvoid}(-1%UInt)))`, but just the code `Ptr{Cvoid}(-1%UInt)`. The fact that we construct and inline that pointer in the IR is just something the compiler does. Perhaps we need to prevent inference from creating `Const` `Ptr`s?

Reply:
yeah, stopping inference from embedding pointers would probably be a better plan